documentation/modules/exploit/linux/http/ivanti_csa_unauth_rce_cve_2021_44529.md

```diff
@@ -0,0 +1,174 @@
+## Vulnerable Application
+
+Ivanti Cloud Services Appliance for Ivanti Endpoint Manager is a appliance that is
+designed to manage endpoints (Desktops). It also know under the name LANDESK. The
+appliance can be either a physical or a virtual appliance and it runs a web based application
+where the HTTP web interface is typically exposed to the public internet.
+
+A code injection vulnerability in the Ivanti EPM Cloud Services Appliance (CSA) before
+version `4.6.0-512` allows an unauthenticated user to execute arbitrary code with limited
+permissions by sending a specially crafted cookie to the client endpoint at `/client/index.php`.
+Successful exploitation results in command execution as user `nobody`.
+
+Installing a vulnerable test bed requires an Ivanti EPM Cloud Services Appliance (CSA),
+either physical or virtual with the vulnerable software installed.
+
+This module has been tested against a virtual Ivanti EPM Cloud Services Appliance (CSA)
+with the specifications listed below:
+
+* Ivanti EPM Cloud Services Appliance (CSA)
+* Version: `4.6.0-20211203.1950` -> not vulnerable
+* Remark: Manually added vulnerable code in `/opt/landesk/broker/webroot/lib/csrf-magic.php`
+
+## Verification Steps
+
+### Installation
+Below are the steps to install and setup a vulnerable Ivanti EPM Cloud Services virtual Appliance (CSA).
+
+* Download the CSA 4.6 virtual appliance
+  [ISO](https://download.ivanti.com/product/CSA/46/ldcsa-scsi-csrffix.iso) and follow the
+  instructions [on the form](https://forums.ivanti.com/s/article/How-to-Create-CSA-VM-from-ISO?language=en_US).
+* Add a second network interface on the VM at your hypervisor. This will allow you to run
+  and test the appliance without activation.
+* Start the appliance again and enable the ssh service in the security tab.
+* Login to the system via SSH with the username `admin` (default password = `admin`).
+* Edit `/opt/landesk/broker/webroot/lib/csrf-magic.php` and add just before
+  `// Load user configuration` section the following code:
```

Comment:
Why?

Reply:
well, the Ivanti CSA 4.6 version that you still can download has already the patch. So you need to add the vulnerable code again to test the module.

Reply:
Calling this out within the module docs would probably be a good idea. I think this is a good explanation and also relates to the removal of the `not vulnerable` comment by adding an appropriate explanation as to why its not vulnerable and how you make it vulnerable again.

Reply:
Explanation added with 57b38a879d9233b5a7817c9f55094fa3425adc46